modules/transport-netty4/src/main/java/org/elasticsearch/http/netty4/Netty4HttpPipeliningHandler.java

```diff
@@ -477,6 +478,16 @@ public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) {
         }
     }
 
+    @Override
+    public void userEventTriggered(ChannelHandlerContext ctx, Object evt) throws Exception {
+        if (evt instanceof SslCloseCompletionEvent closeEvent) {
+            if (closeEvent.isSuccess() && ctx.channel().isActive()) {
+                logger.trace("received TLS close_notify, closing connection {}", ctx.channel());
+                ctx.close();
```

Comment:
I have a minor preference to use:

```java
ctx.channel().close();
```

to ensure that the close initiates from the tail of the pipeline (should we added handlers after this in the future).